In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from collections import Counter

In [2]:
dataset=pd.read_csv("Dataset_for_Internship.csv",encoding='unicode_escape')
dataset.head(5)

Restaurant ID                Restaurant Name  Country Code       City  \
0       17284302  El Vaquero Mexican Restaurant           216     Albany   
1       17330155                       Deorio's           216   Columbus   
2       17334679                         Azteca           216  Davenport   
3       17342548  Happy Joe's Pizza & Ice Cream           216    Dubuque   
4       17582499                    Royal Hotel           216  Pocatello   

   Longitude  Latitude                    Cuisines  Average Cost for two  \
0   -84.2194   31.6158                     Mexican                     0   
1   -84.9427   32.4810              Italian, Pizza                     0   
2   -90.5683   41.5699                     Mexican                     0   
3   -90.7264   42.4955  Desserts, Pizza, Ice Cream                     0   
4  -112.0127   42.6192             Pizza, Bar Food                     0   

    Currency Has Table booking Has Online delivery Is delivering now  \
0  Dollar($)                No                  No                No   
1  Dollar($)                No                  No                No   
2  Dollar($)                No                  No                No   
3  Dollar($)                No                  No                No   
4  Dollar($)                No                  No                No   

   Price range  Aggregate rating Rating color Rating text  Votes  
0            1               3.4       Orange     Average     45  
1            1               4.0        Green   Very Good    170  
2            1               4.3        Green   Very Good    167  
3            1               3.5       Yellow        Good     74  
4            1               3.6       Yellow        Good     59

# Task 2:Cuisine Combination

### 1. Identify the most common combinations of cuisines in the dataset.

In [3]:
#Filter out rows where the 'Cuisines' column contains multiple cuisines (separated by commas)
multi_cuisine_entries = dataset[dataset['Cuisines'].str.contains(',')]

#Split the cuisines by commas and convert to tuples
cuisine_combinations = multi_cuisine_entries['Cuisines'].str.split(',').apply(lambda x: tuple(sorted(x)))

#Count the occurrences of each cuisine combination
most_common_combination = Counter(cuisine_combinations).most_common()

#Convert the result into a DataFrame
common_combination_df = pd.DataFrame(most_common_combination, columns=['Cuisine Combination', 'Count'])
common_combination_df.head(11)

Cuisine Combination  Count
0                 ( Chinese, North Indian)    511
1                 ( Mughlai, North Indian)    334
2       ( Chinese,  Mughlai, North Indian)    267
3                      ( Desserts, Bakery)    170
4                      ( Fast Food, Pizza)    131
5                    ( Fast Food, Chinese)    118
6                   ( Street Food, Mithai)    116
7                     ( Fast Food, Bakery)    108
8                 ( North Indian, Chinese)    105
9   ( Chinese,  Continental, North Indian)     83
10                  ( Desserts, Ice Cream)     83

This data highlights the popularity of cuisine combinations in a restaurant dataset. The most popular combination is "Chinese and North Indian," with 511 occurrences, followed by "Mughlai and North Indian" with 334. These combinations suggest a strong preference for North Indian cuisine paired with other popular types like Chinese and Mughlai. The data indicates that fusion or multi-cuisine offerings tend to attract more customers.

### 2. Determine if certain cuisine combinations tend to have higher ratings

In [4]:
#Filter out rows where the 'Cuisines' column contains multiple cuisines (separated by commas)
multi_cuisine_entries = dataset[dataset['Cuisines'].str.contains(',')].copy()  

#Split the cuisines by commas, sort them, and convert to tuples
multi_cuisine_entries.loc[:, 'Cuisine Combination'] = multi_cuisine_entries['Cuisines'].str.split(',').apply(lambda x: tuple(sorted(x)))

#Group by the 'Cuisine Combination' and calculate the average rating
avg_rating_by_combination = multi_cuisine_entries.groupby('Cuisine Combination')['Aggregate rating'].mean().reset_index()

#Sort by average rating in descending order to see which combinations tend to have higher ratings
avg_rating_by_combination_sorted = avg_rating_by_combination.sort_values(by='Aggregate rating', ascending=False)
avg_rating_by_combination_sorted.head(70)

Cuisine Combination  Aggregate rating
504                    ( Caribbean,  Seafood, American)               4.9
1123                               ( Indonesian, Sunda)               4.9
1235                               ( Lucknowi, Mughlai)               4.9
249                    ( Bakery,  Continental, Italian)               4.9
1443                               ( Seafood, Hawaiian)               4.9
...                                                 ...               ...
1064                                 ( French, Seafood)               4.6
441                                   ( Burmese, Asian)               4.6
1243          ( Mediterranean,  Middle Eastern, Indian)               4.6
40    ( American,  Continental,  Italian,  North Ind...               4.6
680        ( Chinese,  Mughlai,  North Indian, Biryani)               4.6

[70 rows x 2 columns]

Popular cuisine combinations like "Chinese, Mughlai, North Indian" and "Chinese, Mughlai, North Indian, Biryani" tend to have high ratings around 4.6, indicating a positive correlation between popularity and good customer satisfaction.
However, some of the highest-rated combinations (e.g., "Caribbean, Seafood, American" and "Lucknowi, Mughlai") are less common but still achieve top ratings of 4.9. This suggests that niche or less frequent combinations can still achieve high customer satisfaction.